src/match.js

```diff
@@ -0,0 +1,118 @@
+const isStar = (char) => char === '*';
+
+const isPlus = (char) => char === '+';
+
+const isQuestion = (char) => char === '?';
+
+const isOperator = (char) => isStar(char) || isPlus(char) || isQuestion(char);
+
+const isLiteral = (ch) => (ch >= "a" && ch <= "z") || (ch >= "A" && ch <= "Z") || (ch >= "0" && ch <= "9");
+
+const isSet = (term) => isOpenSet(term[0]) && isCloseSet(term.slice(-1));
```

Comment:
Here would be better to use startsWith and endsWith methods to make your function more clear
```suggestion
const isSet = (term) => term.startsWith('[') && term.endsWith(']');
```